public/app/plugins/datasource/loki/live_streams.ts

```diff
@@ -42,9 +42,19 @@ export class LiveStreams {
         appendResponseToBufferedData(response, data);
         return [data];
       }),
-      catchError(err => {
-        return throwError(`error: ${err.reason}`);
-      }),
+      retryWhen((attempts: Observable<any>) =>
```

Comment:
This is an excellent candidate for testing using `expect(getStream()).toEmitValuesWith()`. Maybe you could pass webSocket as an argument to the function so you can pass your own Observable when testing?.